### PR TITLE
Allows mustache syntax in numeric filters

### LINKED ISSUE #14198: Global variable can't be used as a filter value if that value is an integer
### Describe the Bug

At the Insights module, if you want to use a global variable specified by any global variable panel, you won't be able to use it at a filter which requires its value to be an integer. 

Take for instance the following example:

https://user-images.githubusercontent.com/5946359/176522029-5ab78e74-dffe-4ffb-a1c4-5c0061cd6d3f.mp4

Here, `serial` (in contrast to `product`) is an integer, and thus, you won't be able to use `{{ serial }}`.

### To Reproduce

Make sure you have a collection with an integer field. `x` Next, try to create a panel which uses this collection. Third, create a filter and try to apply a filter `x equals {{ var }}`. You will see that any string will be ignored here, and thus the value will be discarded, i.e. `x equals ...`.

### Errors Shown

_No response_

### What version of Directus are you using?

9.13.0

### What version of Node.js are you using?

16.13.0

### What database are you using?

Postgres 13

### What browser are you using?

Chrome

### How are you deploying Directus?

PaaS

### PR DESCRIPTION
**Pull request by jamescammarano**
## Description
Before:

https://user-images.githubusercontent.com/5946359/176522029-5ab78e74-dffe-4ffb-a1c4-5c0061cd6d3f.mp4

After:

https://user-images.githubusercontent.com/67079013/176710977-be84523e-1d64-4364-9330-6ad7a5b06230.mov

Note: This will allow mustache syntax across the entire app's use of system-filter even though this syntax is really only valuable to the Insights dashboard. If this is not ideal I can look for a different solution.

Fixes #14198

## Type of Change

- [x] Bugfix
- [ ] Improvement
- [ ] New Feature
- [ ] Refactor / codestyle updates
- [ ] Other, please describe:

## Requirements Checklist

- [ ] New / updated tests are included
- [x] All tests are passing locally
- [x] Performed a self-review of the submitted code

If adding a new feature:

- [ ] Documentation was added/updated

### CODE REVIEW

**Review by rijkvanzanten — COMMENTED** [review]
Am I misreading, or this function calling `emit('input', val)` for every case? If that's true, could this function not just simply be:

```js
function emitValue(val: string) {
  if (val === '') return emit('input', null);
  emit('input', val);
}
```

**Comment by jaycammarano** [comment]
> Am I misreading, or this function calling `emit('input', val)` for every case? If that's true, could this function not just simply be:
> 
> ```js
> function emitValue(val: string) {
>   if (val === '') return emit('input', null);
>   emit('input', val);
> }
> ```

is there a time when it wouldn't match:
```ts

if (typeof val !== 'string' || new RegExp(inputPattern.value).test(val)) {
	return emit('input', val);
}
```

File: app/src/interfaces/_system/system-filter/input-component.vue [hunk 1]
```diff
@@ -152,7 +152,7 @@ export default defineComponent({
 		return { displayValue, width, t, emitValue, inputEl, inputPattern, dateTimeMenu };
 
 		function emitValue(val: unknown) {
-			const mustache = new RegExp(/({{.*?}})/g);
+			const mustache = new RegExp(/^({{.*?}})$/);
```
Comment (licitdev) [inline app/src/interfaces/_system/system-filter/input-component.vue:155]: @rijkvanzanten Should "multi tags" be allowable in the system filter context for insights? Eg: users can form custom strings `{{var1}}_{{var2}}`.

I wonder if a property should be added to `allowVariables` on `system-filter.vue` as it is currently only supported in insights. 🤔

The regex could be updated to along the lines of `({{[\w\d_]+?}})` as only alphanumeric characters and underscores are allowed.

Comment (jaycammarano) [reply app/src/interfaces/_system/system-filter/input-component.vue:155]: The regex should definitely be updated at the very least

Comment (rijkvanzanten) [reply app/src/interfaces/_system/system-filter/input-component.vue]: Seeing we're now using this interface inside insights options as well, it should support the multi-var style too yeah

**Comment by rijkvanzanten** [comment]
Heya, thanks for your PR! Our small team has been attempting to work through a backlog of external contributions for the better part of a year, but in that time many have become too stale to merge or the change is no longer desired/relevant.. 
We've updated the [contributing guidelines](https://github.com/directus/directus/blob/main/contributing.md#pull-requests) and are working on [making more resources available](https://github.com/directus/directus/discussions/17977) to make sure that doesn’t happen again 😬 In the meantime, we'll have to close this PR for now..

**Pull request closed without merging**